### ISSUE TITLE
Unreachable code detected after an IIFE with try-catch.

### ISSUE THREAD

**Issue by proteriax** [microsoft/TypeScript#38043]
**TypeScript Version:**  3.8.3

**Search Terms:**

**Code**

```ts
const success = (() => {
    try {
        doAnything();
        return true;
    } catch {
        return false;
    }
})();

console.log('Reachable') // < Unreachable code detected.(7027)
```

**Expected behavior:**
Last time is reachable.

**Actual behavior:**
Unreachable code detected.(7027)

**Playground Link:** [Link](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBBCuxgFMIRgXhgCmwSkwD4YBvAKBkpigCcBPUiq5gExAEEw6oALASzABzfAG4mzSjWRR4NMNRrxkY5gF8YwAIZRgPRhKpSZcmADNNAGwjLxMVWVV5RZMqEggLyAHQWQwgOQASsiaupoARp7+eEA)

**Related Issues:** 
#37998

**Comment by IllusionMH** [comment]
Already fixed in 3.9 builds
https://github.com/microsoft/TypeScript/issues/36828

**Comment by proteriax** [comment]
Thanks
